### ISSUE TITLE
Can Strongly Typed Convenience Methods be Added?

### ISSUE THREAD

**Issue by AdamDotNet** [mysql-net/MySqlConnector#313]
Hello.

Most/all ADO.Net providers I've used have returned their specific types instead of the `System.Data.Common` types. An example is [PostgreSQL:CreateCommand()](https://github.com/npgsql/npgsql/blob/292e4e958082379e89fab4f68c558426d29f976c/src/Npgsql/NpgsqlConnection.cs#L457).

Are you opposed to these convenience methods?

Taking a glance at 0.25.1 code, seems like, for example, it could be changed from:

```CSharp
protected override DbCommand CreateDbCommand() => new MySqlCommand(this, CurrentTransaction);
```

To:
```CSharp
/// <summary>
/// TODO: Documentation
/// </summary>
/// <returns></returns>
public new MySqlCommand CreateCommand() => new MySqlCommand(this, CurrentTransaction);

protected override DbCommand CreateDbCommand() => CreateCommand();
```

It would be good to apply these kinds of changes to all applicable `DbConnection`, `DbCommand`, etc. methods as well. This would make provider specific, and helpful, APIs become more visible and reduce friction from always having to cast from the common types. Example: #127, you can't see `AddWithValue` unless you cast.

**Comment by bgrainger** [comment]
Here are the "`public new`" methods in Connector/NET, which could be cloned for API compatibility (ticks by the ones that already exist):

```
MySqlCommand:
public new MySqlConnection Connection { get; set; }
public new MySqlParameterCollection Parameters { get; } ✔️
public new MySqlTransaction Transaction { get; set; }
public new MySqlParameter CreateParameter()
public new MySqlDataReader ExecuteReader()
public new MySqlDataReader ExecuteReader(CommandBehavior)

MySqlConnection:
public new MySqlTransaction BeginTransaction() ✔️
public new MySqlTransaction BeginTransaction(IsolationLevel)
public new MySqlCommand CreateCommand()

MySqlDataReader:
public new void Dispose() 😱

MySqlParameterCollection:
public new MySqlParameter this[int index] { get; set; } ✔️
public new MySqlParameter this[string name] { get; set; }
```

**Comment by AdamDotNet** [comment]
Awesome, thanks!

**Comment by bgrainger** [comment]
Shipped in [0.26.0](https://www.nuget.org/packages/MySqlConnector/0.26.0).
